### PR TITLE
feat(player): decouple Logic, Starter and PlayerClient

### PR DESCRIPTION
**Pull request by xeruf**
Since this changes the interface for participants, maybe something to postpone a bit, unless we're not planning any more releases this season.

### CODE REVIEW

**Review by anarchuser — APPROVED** [review]
A good example of composition over inheritance. Having Starter and Client separate makes sense

File: player/src/main/sc/player2021/logic/Logic.java [hunk 1]
```diff
@@ -20,49 +20,32 @@
...
-
-  /** {@inheritDoc} */
   public void gameEnded(GameResult data, Team color, String errorMessage) {
     log.info("Das Spiel ist beendet.");
   }
 
-  /** {@inheritDoc} */
   @Override
-  public void onRequestAction() {
+  public Move calculateMove() {
```
Comment (SKoschnicke) [inline player/src/main/sc/player2021/logic/Logic.java:33]: Much nicer name! This will be easier to understand for participants.

**Review by SKoschnicke — APPROVED** [review]
LGTM

**Comment by xerus2000** [comment]
@SKoschnicke can you address the point of whether it makes sense to still merge this for the current season?

**Comment by SKoschnicke** [comment]
Shouldn’t go into into the current season.

**Review by SKoschnicke — APPROVED** [review]
LGTM

**Pull request merged**